hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/pipes/Application.java

```diff
@@ -266,4 +272,32 @@ public static String createDigest(byte[] password, String data)
     return SecureShuffleUtils.hashFromString(data, key);
   }
 
+  private class PingSocketCleaner extends Thread {
+    PingSocketCleaner(String name) {
+      super(name);
+    }
+
+    @Override
+    public void run() {
+      LOG.info("PingSocketCleaner started...");
+      while (true) {
+        Socket clientSocket = null;
+        try {
+          clientSocket = serverSocket.accept();
+          LOG.debug("Connection received from {}", clientSocket.getInetAddress());
+          int readData = 0;
+          while (readData != -1) {
+            readData = clientSocket.getInputStream().read();
+          }
+          LOG.debug("close socket cause client has closed.");
+          clientSocket.close();
+        } catch (IOException exception) {
+          LOG.error("PingSocketCleaner exception", exception);
+          IOUtils.closeSocket(clientSocket);
```

Comment:
Do not need this line. `IOUtils.closeSocket` in the finally clause is always executed.
```suggestion
```